src/operator/contrib/crelu-inl.h

```diff
@@ -0,0 +1,185 @@
+
+/*!
+ * Copyright (c) 2017 by Contributors
+ * \file crelu-inl.h
+ * \brief concat relu
+ * \author Yijie Zhuang
+*/
+
+#ifndef MXNET_OPERATOR_CONTRIB_CRELU_INL_H_
+#define MXNET_OPERATOR_CONTRIB_CRELU_INL_H_
+
+#include <dmlc/logging.h>
+#include <dmlc/parameter.h>
+#include <mxnet/operator.h>
+#include <algorithm>
+#include <map>
+#include <vector>
+#include <string>
+#include <utility>
+#include "../operator_common.h"
+#include "../channel_op_common.h"
+#include "../mshadow_op.h"
+
+namespace mxnet {
+namespace op {
+
+namespace concat_relu {
+enum CReluOpInputs {kData};
+enum CReluOpResource {kTempSpace};
+enum CReluOpOutputs {kOut};
+}
+
+struct CReluParam : public dmlc::Parameter<CReluParam> {
+  DMLC_DECLARE_PARAMETER(CReluParam) {}
+};
+
+template<typename xpu, typename DType>
+class CReluOp : public Operator {
+ public:
+  explicit CReluOp(CReluParam p) { this->param_ = p; }
+
+  virtual void Forward(const OpContext &ctx,
+                       const std::vector<TBlob> &in_data,
+                       const std::vector<OpReqType> &req,
+                       const std::vector<TBlob> &out_data,
+                       const std::vector<TBlob> &aux_args) {
+    using namespace mshadow;
+    using namespace mshadow::expr;
+    CHECK_EQ(in_data.size(), 1U);
+    CHECK_EQ(out_data.size(), 1U);
+    Stream<xpu> *s = ctx.get_stream<xpu>();
+
+    Tensor<xpu, 2, DType> data = in_data[concat_relu::kData].FlatTo2D<xpu, DType>(s);
+    Tensor<xpu, 2, DType> out = out_data[concat_relu::kOut].FlatTo2D<xpu, DType>(s);
+    std::vector<Tensor<xpu, 2, DType> > concat_data(2);
```

Comment:
I saw similar code in `concat-inl.h`, which create a vector of tensors and then assign each of the tensor.
L55 just specify the type of elements, L57 and L58 specify the pointer/memory space for the tensors.